### PR TITLE
Dashboard-Scenes: JSON Model and Version tabs don't hide edit mode on refresh

### LINKED ISSUE #81218: Refreshing JSON Edit settings or Restore dashboard doesn't set edit mode to true

### PR DESCRIPTION
**Pull request by harisrozajac**
Fixed the issue where refreshing JSON Edit settings or Version tabs dashboard hides the edit mode buttons. 

Also refactored the Versions tab to wrap the whole logic under Page and NavToolbarActions to avoid repeating these components for each case.

Fixes #81218 

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

File: public/app/features/dashboard-scene/settings/VersionsEditView.tsx [hunk 1]
```diff
@@ -188,32 +189,30 @@ function VersionsEditorSettingsListView({ model }: SceneComponentProps<VersionsE
...
-            newInfo={newInfo!}
-            baseInfo={baseInfo!}
-            isNewLatest={isNewLatest!}
-            diffData={model.diffData}
-            onRestore={dashboard.onRestore}
-          />
-        )}
-      </Page>
+  const viewModeCompare = (
+    <VersionHistoryHeader
```
Comment (mdvictor) [inline public/app/features/dashboard-scene/settings/VersionsEditView.tsx:193]: I think this needs to be wrapped together, version comparison no longer works.

**Review by mdvictor — CHANGES_REQUESTED** [review]
Nice work! This fixes the issue!

I left a comment for the `viewModeCompare` variable, as comparison doesn't work now due to not having the components wrapped properly.

File: public/app/features/dashboard-scene/settings/VersionsEditView.tsx [hunk 2]
```diff
@@ -188,32 +189,30 @@ function VersionsEditorSettingsListView({ model }: SceneComponentProps<VersionsE
...
+    ) : (
+      <VersionHistoryComparison
+        newInfo={newInfo!}
+        baseInfo={baseInfo!}
+        isNewLatest={isNewLatest!}
+        diffData={model.diffData}
+        onRestore={dashboard.onRestore}
+      />
     );
   }
```
Comment (mdvictor) [inline public/app/features/dashboard-scene/settings/VersionsEditView.tsx:212]:
```suggestion
  const viewModeCompare = (
    <>
      <VersionHistoryHeader
        onClick={model.reset}
        baseVersion={baseInfo?.version}
        newVersion={newInfo?.version}
        isNewLatest={isNewLatest}
      />
      {
        isLoading ? (
          <VersionsHistorySpinner msg="Fetching changes&hellip;" />
        ) : (
          <VersionHistoryComparison
            newInfo={newInfo!}
            baseInfo={baseInfo!}
            isNewLatest={isNewLatest!}
            diffData={model.diffData}
            onRestore={dashboard.onRestore}
          />
        )
      }
    </>
  );
```

**Review by dprokop — APPROVED** [review]
@harisrozajac - this works well. The suggestion by @mdvictor is fixing the issue for version comparison view, so please apply it and we are good to merge :)

File: public/app/features/dashboard-scene/settings/VersionsEditView.tsx (hunk 1 above)
Comment (harisrozajac) [reply public/app/features/dashboard-scene/settings/VersionsEditView.tsx]: Good point! I've fixed this and compare view now works

**Review by mdvictor — APPROVED** [review]

**Pull request merged**